ansible/setup/index.ts

```diff
@@ -0,0 +1,75 @@
+import os from 'os';
+
+import {ok} from 'assert';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as cache from '@actions/cache';
+import {exec} from '@actions/exec';
+
+interface Input {
+  version: string;
+  pipmirror: string;
+}
+
+function getInput(): Input {
+  const version = core.getInput('version');
+  if (version === '') {
+    throw new Error("'version' input must be non-empty");
+  }
+  if (version.startsWith('v')) {
+    throw new Error("'version' input should not be prefixed with 'v'");
+  }
+  const versionRegex =
+    /^([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$/i;
+
+  if (!versionRegex.test(version)) {
+    throw new Error(
+      "incorrect 'version' specified, it should include all parts of the version e.g 11.0.1"
+    );
+  }
+
+  const pipmirror = core.getInput('pipmirror');
+
+  return {
+    version,
+    pipmirror,
+  };
+}
+
+const toolName = 'ansible';
+
+function _getCacheDirectory(): string {
+  const cacheDirectory = process.env['RUNNER_TOOL_CACHE'] || ''
+  ok(cacheDirectory, 'Expected RUNNER_TOOL_CACHE to be defined')
+  return cacheDirectory
+}
+
+async function run(): Promise<void> {
+  const input = getInput();
+  const version = input.version;
+  const pipmirror = input.pipmirror;
+  const cachepath = _getCacheDirectory();
+  const _path = path.join(cachepath, toolName, version);
+  core.debug(`tool path: ${_path}`);
+
+  const key = `tool-${toolName}-${version}`;
+  const hit = await cache.restoreCache([_path], key);
+  if (hit !== undefined) {
+    core.info(`Found tool cache for ${toolName} ${version}`);
+    core.addPath(`${_path}/bin`);
+    return
+  }
+
+  core.info(`Creating virtualenv ...`);
+  await exec(`python3 -m venv ${_path}`);
```

Comment:
do we need to check if `python3` exist on runner before we install ?

Reply:
I think it will be included in runner setup playbook, it will make sure python3-venv is installed in runner hosts.

Reply:
either to setup python in another playbook or with another setup like https://github.com/actions/setup-python, at least  we need to output msg if the python3 is missing

Reply:
in case the workflow which is using this action and not running on self-hosted runners, there is a chance python3 is missing from that env.

currently it seems impossible to make a dependency between actions, unless we duplicate the code into this action.

so we have to add a extra `uses` before using ansible setup action.

```
# some workflow
jobs:
  run-playbook:
  - uses: actions/setup-python
    with:
      python-version: 3.11
  - uses: wiredcraft/github-actions/ansible/setup
    with:
      version: 6.6.0
```

Reply:
we will not do any python setup in this step,  I mean just check if python3 exists before we run `exec(python3 -m venv`, if doesn't exist then just raise a error msg and exit

Reply:
check python3 before executing any commands